library-udf/src/main/java/org/apache/iotdb/library/dprofile/UDTFDistinct.java

```diff
@@ -52,6 +55,8 @@ public class UDTFDistinct implements UDTF {
   private HashSet<String> stringSet;
   private Type dataType;
 
+  private Map<Object, Long> minTimeMemo;
```

Comment:
use `XXXLongHashMap` instead of previous `XXXHashSet` and your newly added `Map<Object, Long>`